### PR TITLE
App Check: Token API for 3P  use

### PR DESCRIPTION
**Pull request by ncooke3**
- Added Public Get Token API for 3P use
- Refactored `AppCheckTests.m` to share testing infra between new public get token API and interop get token API
- Added API-reviewed error enum
- Add error handling for keychain related errors
- Add tests for above addition

b/190653133

### CODE REVIEW

File: FirebaseAppCheck/Sources/Public/FirebaseAppCheck/FIRAppCheck.h
Comment (ncooke3) [inline FirebaseAppCheck/Sources/Public/FirebaseAppCheck/FIRAppCheck.h:66]: Briefly discussed after API review but wanted to touch base about API naming. I've renamed this method from `getAppCheckTokenForcingRefresh` → `appCheckTokenForcingRefresh` to make it more swift async/await friendly without having to additionally specify the `NS_SWIFT_ASYNC_NAME`. I'm thinking the general guidance could be to remove the `get` prefix from any new APIs (if applicable). This comes from Apple's recommendation to remove the `get` prefix from async/await method names.

Swift Async/Await usage example:
```swift
let token = try await AppCheck.appCheck().appCheckToken(forcingRefresh: true)
```
Side note, since "app check" appears three times in the above call, I wonder if `token(forcingRefresh:)` is a better option. 
```swift
let token = try await AppCheck.appCheck().token(forcingRefresh: true)
```

Thoughts?

Comment (maksymmalyhin) [reply FirebaseAppCheck/Sources/Public/FirebaseAppCheck/FIRAppCheck.h:66]: As discussed offline, dropping `get` looks fine. Let's update the API proposal correspondingly and send a heads up to the API council.

**Review by maksymmalyhin — COMMENTED** [review]
👍 LGTM after adding tests.

File: FirebaseAppCheck/Sources/Core/Errors/FIRAppCheckErrorUtil.m
Comment (maksymmalyhin) [inline FirebaseAppCheck/Sources/Core/Errors/FIRAppCheckErrorUtil.m:52]: optional: What if `error` passed is not keychain one? I would check the expected domain/codes and fallback to `FIRAppCheckErrorCodeUnknown`.

File: FirebaseAppCheck/Sources/Core/Errors/FIRAppCheckHTTPError.m
Comment (maksymmalyhin) [inline FirebaseAppCheck/Sources/Core/Errors/FIRAppCheckHTTPError.m:20]: nit: The "main" header that defines the class should be first in the file.

File: FirebaseAppCheck/Tests/Unit/AppAttestProvider/Storage/FIRAppAttestArtifactStorageTests.m
Comment (maksymmalyhin) [inline FirebaseAppCheck/Tests/Unit/AppAttestProvider/Storage/FIRAppAttestArtifactStorageTests.m:18]: nit: I think the headers should be reordered a bit:
```
#import <XCTest/XCTest.h> // 1st group. The "main" required header.

#import <OCMock/OCMock.h> // Another group.

#import <GoogleUtilities/GULKeychainStorage.h> // Another group.
```
See go/objc-style#order-of-includes for more context.

**Review by maksymmalyhin — APPROVED** [review]
LGTM. Thank you for the clean tests!

File: FirebaseAppCheck/Tests/Unit/AppAttestProvider/Storage/FIRAppAttestArtifactStorageTests.m
Comment (ncooke3) [reply FirebaseAppCheck/Tests/Unit/AppAttestProvider/Storage/FIRAppAttestArtifactStorageTests.m:18]: Thank you for pointing this out 👍

**Review by maksymmalyhin — APPROVED** [review]
👍 Thank you for the fixes!

File: FirebaseAppCheck/Sources/Core/Errors/FIRAppCheckErrorUtil.m
Comment (ncooke3) [reply FirebaseAppCheck/Sources/Core/Errors/FIRAppCheckErrorUtil.m]: I moved some things around and now check the domain/codes, falling back to `FIRAppCheckErrorCodeUnknown` if necessary. PTAL

File: FirebaseAppCheck/Sources/Public/FirebaseAppCheck/FIRAppCheck.h
Comment (ncooke3) [inline FirebaseAppCheck/Sources/Public/FirebaseAppCheck/FIRAppCheck.h:66]: This API's signature aligns with the amended API proposal. 

Usage
```swift
// Completion handler
  let token = AppCheck.appCheck().token(forcingRefresh: true) { token, error in
    // ...
  }

// Async await
let token = try await AppCheck.appCheck().token(forcingRefresh: true)
```
```objc
  [[FIRAppCheck appCheck] tokenForcingRefresh:NO
                                   completion:^(FIRAppCheckToken *_Nullable token, NSError *_Nullable error) {
                                      // ...
                                   }];
```

Comment (ncooke3) [reply FirebaseAppCheck/Sources/Public/FirebaseAppCheck/FIRAppCheck.h:66]: I've updated the source to reflect the amended API and notified the API team yesterday (there have been no updates to the thread). 

As a final check before merging, I just wanted to check I've taken required steps before merging a new API. 
cc: @ryanwilson @paulb777

File: FirebaseAppCheck/Sources/Core/FIRAppCheck.m
Comment (paulb777) [inline FirebaseAppCheck/Sources/Core/FIRAppCheck.m:178]: Not necessarily for this PR, but we should start being Swift first in error messages.

File: FirebaseAppCheck/Sources/Public/FirebaseAppCheck/FIRAppCheck.h
Comment (paulb777) [reply FirebaseAppCheck/Sources/Public/FirebaseAppCheck/FIRAppCheck.h:66]: LGTM, it would be great to have a Swift API test.

**Review by paulb777 — APPROVED** [review]

File: FirebaseAppCheck/Sources/Core/FIRAppCheck.m
Comment (ncooke3) [reply FirebaseAppCheck/Sources/Core/FIRAppCheck.m:178]: Will audit repo and issue a follow-up PR with Swift first error messages.

**Pull request merged**
